department-of-veterans-affairs/abd-vro
# Mcp 2008 - Integration BIP Claims API mock service

Integrating BIP mock services

## What was the problem?
BIP API service implemented in abd-vro has no connection to any BIP API services in dev environment. Without a BIP API access in development environment, it is difficult to further the BIP API integration development in abd-vro.

This PR integrates the mock services for BIP API abd-vro will use for abd-vro dev environment. Using the mock service, we can continue the BIP API integration development work in abd-vro development environment.

To test the BIP API service with mock services, a BIP controller is created in this PR. It provides the access to BIP claim API that are considered to be used in abd-vro. There are five endpoints, get-claim, get-contention-list-for-a-claim, set-claim-to-RFD-status, and update-contention. In addition, an additional endpoint, create-a-contention, is provided in BIP controller. However, the mock service has not been able to work correctly for this API. Since this API, create-contention, is not required for abd-vor now. We can remove it later. I keep it here in case the 3rd party's mock service we are using fixes the problem and we need it again in the near future.

The BIP controller can be removed later. It is here just for an easy way to verify the BIP service.

NOTE: 
This PR disabled 3 tests in app. These tests failed after mock service is integrated. One is due to the expected result cannot be provided by mock service. The other two are related to camel tests. I suggest to create new Jira ticket(s) to address them.

Associated tickets or Slack threads:
MCP-2008

## How does this fix it?[^1]
Connects to mock services in development environment.

## How to test this PR
- Step 1. Update environment variables by run updated setenv.sh in secret repo
- Step 2[^secrel]. Build application and bring up the swagger UI.
- Step 3. Go to BIP Controller, test the endpoints using claim ID, 9729039


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
